calculator_test.go

```diff
@@ -2,23 +2,332 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
 )
 
+type testCase struct {
+	a, b        float64
+	want        float64
+	name        string
+	errExpected bool
+}
+
 func TestAdd(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
+	testCases := []testCase{
+		{a: 1, b: 1, want: 2, name: "Add 1 + 1"},
+		{a: 10.0, b: -2, want: 8.0, name: "Add a positive and a negative"},
+		{a: 5, b: 3, want: 8, name: "Add 5 + 3"},
+	}
+	for _, tc := range testCases {
+		got := calculator.Add(tc.a, tc.b)
+		if tc.want != got {
+			t.Errorf("Error for the following test: ")
```

Comment:
Do your test output in one `t.Errorf` call, instead of three. Instead of printing the test case name, pass it to `t.Run` and use that to run the test case.